### PR TITLE
bluetooth: host: fix few mismatched CONTAINER_OF

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Fix a bunch of mismatched CONTAINER_OF, few missing k_work_delayable_from_work conversions but also many bt_l2cap_le_chan/bt_l2cap_chan and few others.

### CODE REVIEW

File: subsys/bluetooth/host/l2cap.c [hunk 1]
```diff
@@ -911,10 +912,11 @@ static int l2cap_chan_le_send_sdu(struct bt_l2cap_le_chan *ch,
 
 static void l2cap_chan_tx_process(struct k_work *work)
 {
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
 	struct bt_l2cap_le_chan *ch;
 	struct net_buf *buf;
 
-	ch = CONTAINER_OF(work, struct bt_l2cap_le_chan, tx_work);
+	ch = CONTAINER_OF(dwork, struct bt_l2cap_le_chan, tx_work);
```
Comment (Thalley) [inline subsys/bluetooth/host/l2cap.c:919]: Could do 

```suggestion
	ch = CONTAINER_OF(k_work_delayable_from_work(work), struct bt_l2cap_le_chan, tx_work);
```

as `dwork` isn't used for any other purpose

File: tests/bsim/bluetooth/host/l2cap/stress/src/main.c [hunk 2]
```diff
@@ -178,7 +178,8 @@ static struct bt_l2cap_chan_ops ops = {
 
 void deferred_send(struct k_work *item)
 {
-	struct test_ctx *ctx = CONTAINER_OF(item, struct test_ctx, work_item);
+	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
+	struct test_ctx *ctx = CONTAINER_OF(dwork, struct test_ctx, work_item);
```
Comment (Thalley) [inline tests/bsim/bluetooth/host/l2cap/stress/src/main.c:182]:
```suggestion
	struct test_ctx *ctx = CONTAINER_OF(k_work_delayable_from_work(item), struct test_ctx, work_item);
```

**Review by Thalley — DISMISSED** [review]
Nice catches. Looks like we could expand the `CONTAIN_OF` macros that L2CAP already use to make some of these simpler, but that could be a new PR

File: subsys/bluetooth/host/l2cap.c (hunk 1 above)
Comment (fabiobaltieri) [reply subsys/bluetooth/host/l2cap.c]: Done

**Review by Thalley — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
> Nice catches. Looks like we could expand the `CONTAIN_OF` macros that L2CAP already use to make some of these simpler, but that could be a new PR

Yeah these could probably use some more helpers I guess, want to pick up the patch and submit a different version yourself? You can cherry pick https://github.com/zephyrproject-rtos/zephyr/pull/61962/commits/1bcd92d39bfcfee3c3b3b215087fd3d634bc6bc1 to test it. Or submit this and follow up, either works for me.

**Comment by Thalley** [comment]
> > Nice catches. Looks like we could expand the `CONTAIN_OF` macros that L2CAP already use to make some of these simpler, but that could be a new PR
> 
> Yeah these could probably use some more helpers I guess, want to pick up the patch and submit a different version yourself? You can cherry pick [1bcd92d](https://github.com/zephyrproject-rtos/zephyr/commit/1bcd92d39bfcfee3c3b3b215087fd3d634bc6bc1) to test it. Or submit this and follow up, either works for me.

I think improving these has too low priority for me compared to my other tasks, so hopefully someone else agrees and has the time to do it :D

**Review by jhedberg — APPROVED** [review]

**Pull request merged**